src/nft/utils/txRoute/combineItemsWithTxRoute.ts

```diff
@@ -74,27 +77,33 @@ const itemInRouteAndSamePool = (
 export const combineBuyItemsWithTxRoute = (
   items: UpdatedGenieAsset[],
   txRoute?: RoutingItem[]
-): UpdatedGenieAsset[] => {
-  return items.map((item) => {
+): { hasPriceAdjustment: boolean; updatedAssets: UpdatedGenieAsset[] } => {
+  let hasPriceAdjustment = false
+  const updatedAssets = items.map((item) => {
     const route = getRouteForItem(item, txRoute)
 
-    // if the item is not found in txRoute, it means it's no longer for sale
     if (txRoute && !route) {
       return {
         ...item,
         isUnavailable: true,
       }
     }
 
-    const newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
+    let newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
```

Comment:
it almost seems like if `route.assetOut.priceInfo` exists, newPriceInfo should be set to it no matter what, otherwise do `item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo`

Reply:
that is actually what is happening and regarding the above comment that is why I didn't do it based on if hasPrice diff because if `route.assetOut.priceinfo` exists `newPriceInfo` will be set to it